.github/workflows/validate-pr.yaml

```diff
@@ -129,3 +129,61 @@ jobs:
       - name: Run npm format
         run: |
           cd web && npm run format
+  run-k8s-deployment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Start minikube
+        uses: meroxa/setup-minikube@master
+        with:
+          minikube-version: 1.24.0
+      - name: run k8s deployment on the cluster
+        run: |
+          namespace=monocle
+          kubectl create namespace "${namespace}" 
+          kubectl create --namespace "${namespace}" --filename ./deployment/ 
+
+          deploy_pod_labels_raw=$(kubectl get --namespace "${namespace}" --filename ./deployment/elastic-deployment.yaml --output go-template='{{ range $k, $v :=.spec.selector.matchLabels }}{{$k}}={{$v}},{{ end }}')
+          deploy_pod_labels_elastic=$(echo "${deploy_pod_labels_raw}" | rev | cut --delimiter=',' --fields='2-' | rev)
+
+          while [[ $(kubectl get po --namespace "${namespace}" --no-headers 2>/dev/null | wc -l) -eq  0 ]]; do
+            echo -n "."
+            sleep 1s
+          done
+          echo
+
+          kubectl wait --for=condition=Ready --namespace "${namespace}" pods --selector "${deploy_pod_labels_elastic}" --timeout=1m
+          kubectl wait --for=condition=Ready --namespace "${namespace}" pods --all
+
+          kubectl create --namespace "${namespace}" job test  --from $( kubectl get --namespace ${namespace} -f ./deployment/crawler-legacy-cronjob.yaml --output name) 
+          kubectl wait --for=condition=complete --namespace "${namespace}" job/test 
+
+      - name: Display indexes
+        run: |
+          namespace=monocle
+          kubectl port-forward --namespace "${namespace}" $( kubectl get --namespace "${namespace}" -f deployment/elastic-service.yaml --output name) 9200:9200 &
+          sleep 5s
+
+          curl -s -I -X GET http://localhost:9200/_cat/indices
+
+          kill %1
+
+      # this is commented as on k8s there is no nginx to route traffic
+      - name: Check api service through nginx
+        run: |
+          namespace=monocle
+          kubectl port-forward --namespace "${namespace}" $( kubectl get --namespace "${namespace}" -f deployment/web-service.yaml --output name) 8080:8080 &
```

Comment:
Why do we need that port-forward in the CI test ? (I'm not a k8s expert :p) I mean this port must available publicly to get a working Monocle deployment. Should we include this port-forward directly into the maniftests (if that make sense) ?

Reply:
So we can't, currently the deployment doesn't expose this information externally, thus port forward is required

We can add ingress to allow for external communication (which I think is only required for the webconsole?)

I think we can add this for now and I will add the ingress in another PR (as this one is taking a long time to complete)

Reply:
Ok, let's do that. Any idea about the CI failure https://github.com/change-metrics/monocle/runs/4312366151?check_suite_focus=true ?

Reply:
Yes only the 8080 must be exposed